include/seastar/core/temporary_buffer.hh

```diff
@@ -217,15 +217,15 @@ public:
     ///
     /// \param o buffer to compare with
     /// \return true if and only if contents are the same
-    bool operator==(const temporary_buffer<char>& o) const {
+    bool operator==(const temporary_buffer<CharType>& o) const {
```

Comment:
Already fixed by a ZPP participant as well :) 575f42deb After we rebase, it won't be needed.